common/lib/xmodule/xmodule/js/src/javascript_loader.js

```diff
@@ -0,0 +1,80 @@
+// Generated by CoffeeScript 1.6.1
+(function() {
+    'use strict';
+    this.JavascriptLoader = (function() {
+        function JavascriptLoader() {
+        }
+
+        JavascriptLoader.executeModuleScripts = function(el, callback) {
+            var callbackCalled, completed, completionHandlerGenerator, completionHandlerGeneratorIE,
+                loaded, placeholders;
+            if (!callback) {
+                callback = null; // eslint-disable-line no-param-reassign
+            }
+            placeholders = el.find('.script_placeholder');
+            if (placeholders.length === 0) {
+                if (callback !== null) {
+                    callback();
+                }
+                return [];
+            }
+            completed = (function() {
+                var i, ref, results;
+                results = [];
+                for (i = 1, ref = placeholders.length; ref >= 1 ? i <= ref : i >= ref; ref >= 1 ? ++i : --i) {
+                    results.push(false);
+                }
+                return results;
+            }());
+            callbackCalled = false;
+            completionHandlerGeneratorIE = function(index) {
+                return function() {
+                    if (this.readyState === 'complete' || this.readyState === 'loaded') {
+                        return completionHandlerGenerator(index)();
+                    }
+                    return undefined;
+                };
+            };
+            completionHandlerGenerator = function(index) {
+                return function() {
+                    var allComplete, flag, i, len;
+                    allComplete = true;
+                    completed[index] = true;
+                    for (i = 0, len = completed.length; i < len; i++) {
+                        flag = completed[i];
+                        if (!flag) {
+                            allComplete = false;
+                            break;
+                        }
+                    }
+                    if (allComplete && !callbackCalled) {
+                        callbackCalled = true;
+                        if (callback !== null) {
+                            return callback();
+                        }
+                    }
+                    return undefined;
+                };
+            };
+            loaded = {};
+            return placeholders.each(function(index, placeholder) {
+                var s, src;
+                src = $(placeholder).attr('data-src');
+                if (!(src in loaded)) {
+                    loaded[src] = true;
+                    s = document.createElement('script');
+                    s.setAttribute('src', src);
+                    s.setAttribute('type', 'text/javascript');
+                    s.onload = completionHandlerGenerator(index);
+                    s.onreadystatechange = completionHandlerGeneratorIE(index);
+                    $('head')[0].appendChild(s);
```

Comment:
I'm going to leave this as-is because of the comment (which didn't appear in the original conversion). Your first suggestion most likely works, but I don't know how to test it properly in all browsers.

````
Need to use the DOM elements directly or the scripts won't execute properly.